src/util/SubscriptionPromise.ts

```diff
@@ -0,0 +1,92 @@
+import { ERROR_TIMEOUT } from '../errorhandling/SDKErrors'
+
+/**
+ * A function that determines whether a new incoming value should reject or resolve the promise.
+ *
+ * @param value A new incoming subscription value.
+ * @returns A truthy value if the promise should be rejected/resolved on this value. Return value is used
+ * as the reason in a rejected promise.
+ */
+export interface Evaluator<SubscriptionType> {
+  (value: SubscriptionType): any
```

Comment:
What do you think about
```suggestion
  (value: SubscriptionType): Boolean | any
```

Reply:
was thinking about that... it's a nice hint but would not have any effects. could do though